### DISCUSSION TITLE
Add to the rlgl api just a bit

### DISCUSSION THREAD

**Discussion by gegoggigog** [raysan5/raylib#3560] [Features]
For example ``rlCompileShader`` does not have a corresponding ``detatch``/``deleteShader``. I think it would make sense to add that, so you don't need opengl interop if you want to clean up your shaders. Along the lines of "what the library allocates, the library should be able to delete".

**Comment by Peter0x44** [comment]
Does rlUnloadShaderProgram not do what you expect?
That's what UnloadShader calls...

**Reply by gegoggigog** [reply]
I expect it to call glDeleteProgram, which it does. It does not, however, cal glDeleteShader... which is also what I expected. ;) (i.e., I do not expect it to call glDeleteShader)
My point was that it would be nice to be able to clean up your shaders (not the program (like in rlLoadShaderCode)), without needing to call raw opengl.

Edit: For example when you create compute shaders, you can't rely on rlLoadShaderCode which is used only for vertex and fragment.
